commands/moderation/verbal.js

```diff
@@ -0,0 +1,166 @@
+const Discord = require('discord.js');
+const dateFormat = require('dateformat');
+const {verifyReasonLength} = require('../../helpers/stringHelpers');
+
+module.exports = {
+  name: 'verbal',
+  description: 'sends a user a verbal warning and logs note in db',
+  guildOnly: true,
+  staffOnly: true,
+  minRole: 'Moderator',
+  execute(msg, args, con) {
+    // Make sure only SU, Mods and Admin can run the command
+    const offendingUser =
+      msg.mentions.members.first() || msg.guild.members.cache.get(args[0]);
+
+    if (
+      hasUserTarget(msg, offendingUser) &&
+      notSelf(msg, offendingUser) &&
+      notHighRoller(msg, offendingUser)
+    ) {
+      // Parse the reason for the warning
+      // if no reason provided, return so the bot doesn't go boom
+      const verbalReason = args.slice(1).join(' ');
+      if (!verbalReason) return msg.reply('You need to provide a reason');
+
+      if (!verifyReasonLength(msg.content, msg)) return;
+
+      // Create an embed, craft it, and DM the user
+      try {
+        dmTheUser(msg, offendingUser, verbalReason);
+      } catch (e) {
+        return msg.reply('Unable to send a DM to the specified user.');
+      }
```

Comment:
Right now this isn't working.  The bot still says the verbal was sent because the code isn't async and continues to execute before the catch block is hit.

Both `execute` and `dmTheUser` should be converted to async and we should await line 30 and line 80 (`targetUser.send(embed)`).

I'm also thinking instead of having a custom error message and trying to tailor the catch, we should probably just catch any error and let the Mod know what went wrong, like we did for the ban command.  That way if something new comes up, we'll be aware of it right away.

https://github.com/CodecademyCommunity/codecademy-discord-bot/blob/26ddcb48a0dab76794e847c9599b06962dd09e38/commands/moderation/ban.js#L24-L28